src/main/java/org/bytedeco/javacpp/tools/Generator.java

```diff
@@ -497,6 +498,14 @@ boolean classes(boolean handleExceptions, boolean defineAdapters, boolean conver
         out.println("static jmethodID JavaCPP_stringMID = NULL;");
         out.println("static jmethodID JavaCPP_getBytesMID = NULL;");
         out.println("static jmethodID JavaCPP_toStringMID = NULL;");
+        out.println("#ifdef STRING_BYTES_CHARSET");
+        out.println("#ifdef MODIFIED_UTF8_STRING");
+        out.println("#error \"STRING_BYTES_CHARSET and MODIFIED_UTF8_STRING must not be defined together\"");
```

Comment:
I think we could just have a warning here? I can see MODIFIED_UTF8_STRING having priority over STRING_BYTES_CHARSET for backward compatibility purposes.

Reply:
I don't really see any value in that. This features are mutually exclusive, and to set String charset user will need to add a new define, which means that they most likely wouldn't have any trouble in removing another one at the same time.

Also, C++ standard doesn't specify preprocessor directive for compile warning, only for error. Which means that we would have to add our own that delegates to compiler-specific ones.

Reply:
They are mutually exclusive, but say you have an upstream project and your downstream project is using MODIFIED_UTF8_STRING. For some reason, the upstream project starts using STRING_BYTES_CHARSET, which would now prevent the downstream project from compiling. With only a warning though, both projects can continue to function properly.

Most compilers support the `#warning` directive, and for those that don't, it's just going to end up in an error, so that's fine, right? :)

Reply:
Ah, I see the problem. MSVC is the one that still doesn't support `#warning`. On the other hand, Clang and GCC have been supporting `#pragma message` for a while now, so let's use that. In the worst case, it's only going to end up getting ignored, which is still better than a failed build.